src/main/java/org/opensearch/ad/ml/EntityColdStarter.java

```diff
@@ -404,42 +419,23 @@ private void getEntityColdStartData(String detectorId, Entity entity, ActionList
             ActionListener<Optional<Long>> minTimeListener = ActionListener.wrap(earliest -> {
                 if (earliest.isPresent()) {
                     long startTimeMs = earliest.get().longValue();
-                    nodeStateManager.getAnomalyDetectorJob(detectorId, ActionListener.wrap(jobOp -> {
-                        if (!jobOp.isPresent()) {
-                            listener.onFailure(new EndRunException(detectorId, "AnomalyDetector job is not available.", false));
-                            return;
-                        }
 
-                        AnomalyDetectorJob job = jobOp.get();
-                        // End time uses milliseconds as start time is assumed to be in milliseconds.
-                        // Opensearch uses a set of preconfigured formats to recognize and parse these strings into a long value
-                        // representing milliseconds-since-the-epoch in UTC.
-                        // More on https://tinyurl.com/wub4fk92
-
-                        // Existing samples either predates or coincide with cold start data. In either case,
-                        // combining them without reordering based on time stamps is not ok. We might introduce
-                        // anomalies in the process.
-                        // An ideal solution would be to record time stamps of data points and combine existing
-                        // samples and cold start samples and do interpolation afterwards. Recording time stamps
-                        // requires changes across the board like bwc in checkpoints. A pragmatic solution is to use
-                        // job enabled time as the end time of cold start period as it is easier to combine
-                        // existing samples with cold start data. We just need to appends existing samples after
-                        // cold start data as existing samples all happen after job enabled time. There might
-                        // be some gaps in between the last cold start sample and the first accumulated sample.
-                        // We will need to accept that precision loss in current solution.
-                        long endTimeMs = job.getEnabledTime().toEpochMilli();
```

Comment:
If HC detector realtime job not restarted, the enabled time won't change. If there is no enough data before job enabled time and user don't backfill historical data, there is no chance to pass cold start, right? Seems a critical bug if that's true. We'd better backfill to 1.x too.

Reply:
right. yes, will backfill.